### ISSUE TITLE
Errors when running build.ps1

### ISSUE THREAD

**Issue by Adam25pwc** [pleonex/PokemonConquest#6]
Hey! Sorry to bother, but when I try to run build.ps1, I get this error code:

> Preparing to run build script...
Running build script...
C:/Users/Adam/Desktop/PokemonConquest-master/build.cake:0: Could not find script 'C:/Users/Adam/Desktop/PokemonConquest-master/build.cake'.
Error: Errors occurred while analyzing script.

So I tried instead to run it like this: `.\build.ps1 -script exporter.cake` but it gives another series of errors, most of them being either `error CS0103: The name '***' does not exist in the current context` or `error CS0246: The type or namespace name '***' could not be found (are you missing a using directive or an assembly reference?)`
<details>
<summary>Here's the full output:</summary>

> Preparing to run build script...
Running build script...
error CS0246: The type or namespace name 'AmbitionConquest' could not be found (are you missing a using directive or an assembly reference?)
error CS0246: The type or namespace name 'AmbitionConquest' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(76,40): error CS0246: The type or namespace name 'Font2Binary' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(76,53): error CS0246: The type or namespace name 'FontKind' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(76,63): error CS0103: The name 'FontKind' does not exist in the current context
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(77,39): error CS0246: The type or namespace name 'Font2Yaml' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(79,33): error CS0246: The type or namespace name 'Font2Image' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(79,7): error CS0246: The type or namespace name 'IImage' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(82,40): error CS0246: The type or namespace name 'Font2Binary' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(82,53): error CS0246: The type or namespace name 'FontKind' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(82,63): error CS0103: The name 'FontKind' does not exist in the current context
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(83,39): error CS0246: The type or namespace name 'Font2Yaml' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(85,33): error CS0246: The type or namespace name 'Font2Image' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(85,7): error CS0246: The type or namespace name 'IImage' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(88,42): error CS0246: The type or namespace name 'Font2Binary' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(88,55): error CS0246: The type or namespace name 'FontKind' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(88,65): error CS0103: The name 'FontKind' does not exist in the current context
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(89,39): error CS0246: The type or namespace name 'Font2Yaml' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(91,33): error CS0246: The type or namespace name 'Font2Image' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(91,7): error CS0246: The type or namespace name 'IImage' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(100,24): error CS0246: The type or namespace name 'Binary2PokemonList' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(101,24): error CS0246: The type or namespace name 'PokemonList2Po' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(106,24): error CS0246: The type or namespace name 'Binary2TextList' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(106,41): error CS0246: The type or namespace name 'TextListKind' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(106,55): error CS0103: The name 'TextListKind' does not exist in the current context
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(111,24): error CS0246: The type or namespace name 'Binary2TextList' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(111,41): error CS0246: The type or namespace name 'TextListKind' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(111,55): error CS0103: The name 'TextListKind' does not exist in the current context
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(116,24): error CS0246: The type or namespace name 'Binary2TextList' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(116,41): error CS0246: The type or namespace name 'TextListKind' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(116,55): error CS0103: The name 'TextListKind' does not exist in the current context
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(121,24): error CS0246: The type or namespace name 'Binary2TextList' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(121,41): error CS0246: The type or namespace name 'TextListKind' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(121,55): error CS0103: The name 'TextListKind' does not exist in the current context
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(126,24): error CS0246: The type or namespace name 'Binary2TextList' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(126,41): error CS0246: The type or namespace name 'TextListKind' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(126,55): error CS0103: The name 'TextListKind' does not exist in the current context
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(131,24): error CS0246: The type or namespace name 'Binary2TextList' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(131,41): error CS0246: The type or namespace name 'TextListKind' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(131,55): error CS0103: The name 'TextListKind' does not exist in the current context
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(136,24): error CS0246: The type or namespace name 'Binary2TextList' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(136,41): error CS0246: The type or namespace name 'TextListKind' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(136,55): error CS0103: The name 'TextListKind' does not exist in the current context
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(141,24): error CS0246: The type or namespace name 'Binary2TextList' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(141,41): error CS0246: The type or namespace name 'TextListKind' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(141,55): error CS0103: The name 'TextListKind' does not exist in the current context
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(151,24): error CS0246: The type or namespace name 'Binary2Blocks' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(154,29): error CS0246: The type or namespace name 'Binary2BlockMessages' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(155,28): error CS0246: The type or namespace name 'BlockMessages2Po' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(166,41): error CS0246: The type or namespace name 'Message' could not be found (are you missing a using directive or an assembly reference?)
Error: Error(s) occurred when compiling build script:
error CS0246: The type or namespace name 'AmbitionConquest' could not be found (are you missing a using directive or an assembly reference?)
error CS0246: The type or namespace name 'AmbitionConquest' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(76,40): error CS0246: The type or namespace name 'Font2Binary' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(76,53): error CS0246: The type or namespace name 'FontKind' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(76,63): error CS0103: The name 'FontKind' does not exist in the current context
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(77,39): error CS0246: The type or namespace name 'Font2Yaml' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(79,33): error CS0246: The type or namespace name 'Font2Image' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(79,7): error CS0246: The type or namespace name 'IImage' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(82,40): error CS0246: The type or namespace name 'Font2Binary' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(82,53): error CS0246: The type or namespace name 'FontKind' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(82,63): error CS0103: The name 'FontKind' does not exist in the current context
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(83,39): error CS0246: The type or namespace name 'Font2Yaml' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(85,33): error CS0246: The type or namespace name 'Font2Image' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(85,7): error CS0246: The type or namespace name 'IImage' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(88,42): error CS0246: The type or namespace name 'Font2Binary' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(88,55): error CS0246: The type or namespace name 'FontKind' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(88,65): error CS0103: The name 'FontKind' does not exist in the current context
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(89,39): error CS0246: The type or namespace name 'Font2Yaml' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(91,33): error CS0246: The type or namespace name 'Font2Image' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(91,7): error CS0246: The type or namespace name 'IImage' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(100,24): error CS0246: The type or namespace name 'Binary2PokemonList' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(101,24): error CS0246: The type or namespace name 'PokemonList2Po' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(106,24): error CS0246: The type or namespace name 'Binary2TextList' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(106,41): error CS0246: The type or namespace name 'TextListKind' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(106,55): error CS0103: The name 'TextListKind' does not exist in the current context
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(111,24): error CS0246: The type or namespace name 'Binary2TextList' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(111,41): error CS0246: The type or namespace name 'TextListKind' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(111,55): error CS0103: The name 'TextListKind' does not exist in the current context
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(116,24): error CS0246: The type or namespace name 'Binary2TextList' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(116,41): error CS0246: The type or namespace name 'TextListKind' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(116,55): error CS0103: The name 'TextListKind' does not exist in the current context
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(121,24): error CS0246: The type or namespace name 'Binary2TextList' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(121,41): error CS0246: The type or namespace name 'TextListKind' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(121,55): error CS0103: The name 'TextListKind' does not exist in the current context
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(126,24): error CS0246: The type or namespace name 'Binary2TextList' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(126,41): error CS0246: The type or namespace name 'TextListKind' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(126,55): error CS0103: The name 'TextListKind' does not exist in the current context
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(131,24): error CS0246: The type or namespace name 'Binary2TextList' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(131,41): error CS0246: The type or namespace name 'TextListKind' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(131,55): error CS0103: The name 'TextListKind' does not exist in the current context
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(136,24): error CS0246: The type or namespace name 'Binary2TextList' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(136,41): error CS0246: The type or namespace name 'TextListKind' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(136,55): error CS0103: The name 'TextListKind' does not exist in the current context
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(141,24): error CS0246: The type or namespace name 'Binary2TextList' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(141,41): error CS0246: The type or namespace name 'TextListKind' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(141,55): error CS0103: The name 'TextListKind' does not exist in the current context
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(151,24): error CS0246: The type or namespace name 'Binary2Blocks' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(154,29): error CS0246: The type or namespace name 'Binary2BlockMessages' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(155,28): error CS0246: The type or namespace name 'BlockMessages2Po' could not be found (are you missing a using directive or an assembly reference?)
C:/Users/Adam/Desktop/PokemonConquest-master/exporter.cake(166,41): error CS0246: The type or namespace name 'Message' could not be found (are you missing a using directive or an assembly reference?)

</details>

By the way, I'm on Windows 10 (64bit). Thanks in advance :)

**Comment by pleonex** [comment]
It seems I am missing some information in the README file. The first time you need to run this command from the folder _AmbitionConquest_: `dotnet build`

Then you can run again the `build.ps1` with the `-script` argument.

**Comment by Adam25pwc** [comment]
Thank you for the quick answer!
I managed to make it run to the "Setup", but now I got another error when I get to the "Open-Game":

> An error occurred when executing task 'Open-Game'.
> Could not find a part of the path 'C:\Users\Adam\Desktop\PokemonConquest-master\GameData\root'.

Which is totally understandable, since I in fact don't have that folder. So I proceed to create it, and I put the root folder in it with the extracted data from Tinke and retry. This time, I get past "Open-Game", but I get stuck on the "Export-Font" task:

> An error occurred when executing task 'Export-Font'.
Object reference not set to an instance of an object.

Now, I already saw the closed issues and found this one: #3 , where the user had the same error but on a different task (the 'Export-TextList' one), but I still decided to try to fix the problem based on your suggestions on that issue, so I decompressed the overlay files with CUE's tool and also extracted the data folder into root\rom, but I still get the same error on the same task. Also, the fact that I didn't have the GameData folder, makes me think I skipped some steps. Thank you again for your time. :)
